### PR TITLE
Add website for `atmos` documentation

### PR DESCRIPTION
**Pull request by aknysh**
## what
* Add website for `atmos` documentation (uses [docusaurus](https://docusaurus.io/docs))

## why
* This PR adds a skeleton for the website for `atmos` documentation
* Supports themes (light and dark) - selected from the header
* Supports client-side search using Algolia
* Supports menus with sections, header/footer with links
* The "Guide" tab will have description on how to configure and use `atmos` and all its features
* The "CLI" tab contains description of all `atmos` commands (the pages are auto-generated from a JSON file describing the CLI commands)
* Each doc section is written in Markdown or [MDX](https://mdxjs.com/), which allows executing React programs using JSX in markdown content. You can import components, such as interactive charts or alerts, and embed them within the content

## TODO
* The menu items will be revisited/reviewed in the consecutive PRs
* All sections will be populated with description in the consecutive PRs
* Add GitHub action to build the site and upload it to AWS S3 + CloudFront
* Host on https://atmos.tools

<br>
image

<br>
image

<br>
image

<br>
image

### CODE REVIEW

File: cmd/aws_eks_update_kubeconfig.go [hunk 1]
```diff
@@ -14,19 +14,19 @@ var awsEksCmdUpdateKubeconfigCmd = &cobra.Command{
 	Long: `This command executes 'aws eks update-kubeconfig' in three different ways:
 
 1. If all the required parameters (cluster name and AWS profile/role) are provided on the command-line, 
-then 'atmos' executes the command without requiring atmos CLI config and context.
+then 'atmos' executes the command without requiring the 'atmos'' CLI config and context.
```
Comment (nitrocode) [inline cmd/aws_eks_update_kubeconfig.go:17]:
```suggestion
then 'atmos' executes the command without requiring the 'atmos.yaml' CLI config and context.
```

File: cmd/aws_eks_update_kubeconfig.go [hunk 2]
```diff
@@ -14,19 +14,19 @@ var awsEksCmdUpdateKubeconfigCmd = &cobra.Command{
...
 For example: atmos aws eks update-kubeconfig --profile=<profile> --name=<cluster_name>
 
 2. If 'component' and 'stack' are provided on the command-line, 
-   then 'atmos' executes the command using atmos CLI config and stack's context by searching for the following settings:
-  - 'components.helmfile.cluster_name_pattern' in 'atmos.yaml' CLI config (and calculates the '--name' parameter using the pattern)
-  - 'components.helmfile.helm_aws_profile_pattern' in 'atmos.yaml' CLI config (and calculates the '--profile' parameter using the pattern)
-  - 'components.helmfile.kubeconfig_path' in 'atmos.yaml' CLI config
+   then 'atmos' executes the command using the 'atmos'' CLI config and stack's context by searching for the following settings:
```
Comment (nitrocode) [inline cmd/aws_eks_update_kubeconfig.go:21]:
```suggestion
   then 'atmos' executes the command using the 'atmos' CLI config and stack's context by searching for the following settings:
```

File: website/docs/README.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# Docs folder
+
+The `/docs` folder houses all markdown files. 
+The page structure loosely maps to the routing on the site since paths can be changed in the frontmatter.
+
+## Versioning
+
+This folder can also contain components, assets, and whatever else is meant to be versioned when the docusaurus versioning script is run. For example,
+if there is a page component that is only relevant to the `layout` section in the current version, it could be added to a `_components/`
+folder in `docs/layout/`. When the versioning script is run, the component will be copied to `versioned_docs/verion-{X}/layout/_components/` and there
```
Comment (nitrocode) [inline website/docs/README.md:10]:
```suggestion
folder in `docs/layout/`. When the versioning script is run, the component will be copied to `versioned_docs/version-{X}/layout/_components/` and there
```

File: website/docusaurus.config.js [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+      trackingID: 'GTM-WQWH2XV',
+    },
+    prism: {
+      theme: {plain: {}, styles: []},
+      // https://github.com/FormidableLabs/prism-react-renderer/blob/master/src/vendor/prism/includeLangs.js
+      additionalLanguages: ['shell-session', 'http'],
+    },
+    algolia: {
+      appId: 'UZ3HBXELUD',
+      apiKey: '1568ce161ef1c20cb929a12dd33df7be',
```
Comment (nitrocode) [inline website/docusaurus.config.js:163]: Is this ok ?

File: website/package.json [hunk 5]
```diff
@@ -0,0 +1,69 @@
+{
```
Comment (nitrocode) [inline website/package.json:1]: Where is the package lock ?

File: website/scripts/bak/build-all/index.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import Listr from 'listr';
```
Comment (nitrocode) [inline website/scripts/bak/build-all/index.ts:1]: Can all this `bak` directory be removed ?

Comment (aknysh) [reply website/scripts/bak/build-all/index.ts:1]: no, it's from [docusaurus](https://docusaurus.io/docs)

Comment (aknysh) [reply website/scripts/bak/build-all/index.ts:1]: it's has code used to generate pages and menues

**Review by nitrocode — APPROVED** [review]

File: website/package.json (hunk 5 above)
Comment (aknysh) [reply website/package.json:1]: not sure if we need to commit it (there are diff opinions, same as terraform lock), but added

**Review by nitrocode — APPROVED** [review]

**Pull request merged**